pandas/core/arrays/interval.py

```diff
@@ -516,10 +517,10 @@ def astype(self, dtype, copy=True):
             else:
                 return self
         elif is_categorical_dtype(dtype):
-            return Categorical(self.values)
+            return Categorical(np.asarray(self))
```

Comment:
Not top priority, but this will result in conversion to interval objects and then in categorical back to IntervalIndex. So long term maybe Categorical should be able to handle arrays natively (so you could just pass `self` here) ?